tests/main.nf.test

```diff
@@ -8,30 +8,41 @@ nextflow_workflow {
 
         when {
             params {
-                input = "samplesheet-staph.csv"
-                platform = "illumina"
+                input = "https://github.com/phac-nml/mikrokondo/raw/integration-testing/tests/data/samplesheets/samplesheet-campy-staph.csv"
                 outdir = "results"
-                mash.mash_sketch = "campy-staph-ecoli.msh"
-                r_contaminants.mega_mm2_idx = "campy.mmi"
+
+                platform = "illumina"
+
+                mash { mash_sketch = "https://github.com/phac-nml/mikrokondo/raw/integration-testing/tests/data/databases/campy-staph-ecoli.msh" }
+                r_contaminants { mega_mm2_idx = "https://github.com/phac-nml/mikrokondo/raw/integration-testing/tests/data/databases/campy.mmi" }
+
                 skip_bakta = true
                 skip_staramr = true
                 skip_mobrecon = true
                 skip_checkm = true
                 skip_raw_read_metrics = true
                 skip_polishing = true
-                max_memory = "4.GB"
-                max_cpus = 2
             }
             workflow {
                 """
-                // define inputs of the workflow here. Example:
-                // input[0] = file("test-file.txt")
+                // define inputs of the workflow here.
                 """
             }
         }
 
         then {
             assert workflow.success
+            assert path("$launchDir/results").exists()
+
+            // parse output json file
+            def json_string = path("$launchDir/results/SummaryReport/final_report.json").readLines().join("\n")
+            def json_parser = new groovy.json.JsonSlurper()
+            def json = json_parser.parseText(json_string)
+
+            assert json.CSE.CSE.FastP.summary.sequencing.equals("paired end (150 cycles + 150 cycles)")
+            assert json.CSE.CSE.FastP.summary.before_filtering.total_reads.equals(248)
+            assert json.CSE.CSE.FastP.filtering_result.passed_filter_reads.equals(248)
+            assert json.CSE.CSE.FastP.insert_size.peak.equals(196)
```

Comment:
Thank you for adding all these asserts.

I noticed there's no assembly QC information in the JSON report. Do you know if it runs SPAdes to create the assembly? Could you add an assert here to check for the existence of the contigs?

We can then maybe work through what other parameters might need to be adjusted to include e.g., assembly quality information in the report.